app/controllers/api/v1/orders_controller.rb

```diff
@@ -123,6 +118,18 @@ def order_response(records)
           exclude_stockit_set_item: true).as_json
       end
 
+      def update_transition_and_cancel_reason
+        event = params['transition'].to_sym
+        @order.update_transition_and_reason(event, cancel_params) if @order.state_events.include?(event)
```

Comment:
Good idea to move the code into the model - logic like this should be in the model so that it can be used elsewhere (such as rake tasks and other classes.) controllers are about transactional flow and models are about data so you've definitely made the right choice here.

Now that this method is only 2 lines, I would put it back up in the `transition` method so that there are fewer jumps to make when reading the code to see what is going on. (Same comment as above)